### DISCUSSION TITLE
Improvement - prevent unnecessary records in ps_image_lang

### DISCUSSION THREAD

**Discussion by samberrry** [PrestaShop/PrestaShop#40146] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

current PrestaShop behavior: 
- when we add images to a product, the multi lang system creates records for each language (just like any other multi lang model derived from ObjectModel)
- when we do not provide a caption for an image a record with empty legend will be added
- perfectly the empty legends is handled in front side with product name for alt tags
- consider a shop with 10,000 products and 2 languages
- normally a product has 3 images on average
- finally we will have 60,000 records in this table
- There will be many empty and useless fields in db, which can be prevented from being inserted

### Describe the solution you'd like

- a solution can be preventing any insertion into the lang table that does not have a value. For example we will prevent the addition of this record: 

| id_{model}  | id_lang | {field} |
| ------------- | ------------- | ------------- |
| 3  | 2 |   |
- this can be configurable through a field in $definition array 
- this way it does not affect other models
- an image without lang records works very well in front
- but in BO the image disappears. I've not checked.. but this can be handled (if it does not have conflict with other parts)
- I do not know if there is any other consideration? I can help for a PR

May be it's not a big issue, but this table is being used in many queries and can be a small improvement.
Thanks guys.

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by florine2623** [comment]
Hello @samberrry ,

Thank you for your suggestion. The Developer Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you or anyone else is willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

Thank you
